### PR TITLE
HTCONDOR-2475 yet-more-code-cleanup

### PR DESCRIPTION
**Pull request by JaimeFrey**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_starter.V6.1/vanilla_proc.cpp [hunk 1]
```diff
@@ -628,16 +615,16 @@ VanillaProc::StartJob()
 							return FALSE;
 						}
 						dprintf(D_FULLDEBUG, "Adding mapping: %s -> %s.\n", full_dir, next_dir);
```
Comment (GregThain) [inline src/condor_starter.V6.1/vanilla_proc.cpp:617]: I think the `next_dir` needs to be `next_dir.c_str()`

**Review by GregThain — CHANGES_REQUESTED** [review]

File: src/condor_daemon_client/daemon.cpp [hunk 2]
```diff
@@ -60,6 +60,7 @@ Daemon::common_init() {
 	Sock::set_timeout_multiplier( param_integer(buf, param_integer("TIMEOUT_MULTIPLIER", 0)) );
 	dprintf(D_DAEMONCORE, "*** TIMEOUT_MULTIPLIER :: %d\n", Sock::get_timeout_multiplier());
 	m_has_udp_command_port = true;
+	daemon_list_it = daemon_list.begin();
```
Comment (GregThain) [inline src/condor_daemon_client/daemon.cpp:63]: I know this is not your doing, but it wasn't intuitively obvious to me on first reading of this code that `daemon_list` really is `collector_list`.  As long as you have this code in your head, should we change the name of this variable, and the related iterator?

**Review by GregThain — CHANGES_REQUESTED** [review]

File: src/condor_starter.V6.1/vanilla_proc.cpp (hunk 1 above)
Comment (JaimeFrey) [reply src/condor_starter.V6.1/vanilla_proc.cpp]: Done

File: src/condor_daemon_client/daemon.cpp (hunk 2 above)
Comment (JaimeFrey) [reply src/condor_daemon_client/daemon.cpp:63]: We should.

**Review by GregThain — APPROVED** [review]
LGTM -- waiting on batlab for most recent change.

**Pull request merged**